captcha/events.py

```diff
@@ -53,10 +60,129 @@ async def cleaner(self, member: Member):
         finally:
             await self.delete_challenge_for(member)
 
+    async def skip_challenge(self, author: Member, challenge: Challenge):
+        roles = [
+            challenge.guild.get_role(role)
+            for role in await self.data.guild(challenge.guild).autoroles()
+        ]
+        try:
+            await self.congratulation(challenge, roles)
+            await self.remove_temprole(challenge)
+
+            def ok_check(msg: str):
+                return f"✅ {msg}"
+
+            await self.send_or_update_log_message(
+                challenge.guild,
+                ok_check(f"Captcha skiped by {author.name} ({author.id})."),
+                challenge.messages["logs"],
+                allowed_tries=(challenge.trynum, challenge.limit),
+                member=challenge.member,
+            )
+            await self.send_or_update_log_message(
+                challenge.guild,
+                bold("Roles added, Captcha skiped."),
+                challenge.messages["logs"],
+                member=challenge.member,
+            )
+        except MissingPermissions:
+            roles_name = [role.name for role in roles]
+            try:
+                await challenge.member.send(
+                    f"Please contact the administrator of {challenge.guild.name} for obtaining "
+                    "access of the server, I was unable to add you the roles on the server.\nYou "
+                    f"should have obtained the following roles: "
+                    f"{humanize_list(roles_name) if roles_name else 'None.'}"
```

Comment:
```suggestion
                    f"Please contact the administrator of {challenge.guild.name} in order to obtain "
                    "access to the server, I was unable to give you the roles on the server."
```
Don't need to list roles here

Reply:
It was already in the original, but I'm taking it out.